haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import warnings
+from collections import defaultdict
+from typing import List, Union, Optional, Tuple, Dict
+
+from dateutil.parser import parse, ParserError
```

Comment:
Hey just to be aware the library `dateutil` might be an additional dependency we add to Haystack. It does not come automatically included with base python (so it is not a standard library). However, it is possible is installed by other packages in Haystack as a secondary dependency. 

Either way it would be smart to check if this node works with a minimal install of haystack (with no extra dependencies installed). If it does not then we will most likely need to add `dateutil` to our dependencies.

Reply:
I've tested it in a virtual environment w/a basic Haystack installation (pip install farm-haystack) and it worked, didn't require to install dateutil separately - indeed seems like it was installed as a secondary dependancy. So, I guess no need to add it to the dependencies file 👌